Rewrite README to be consistent with other apps (#2362)
alphagov/collections: README.md

- **Sub-topic**: a group of content within a topic. (A deprecated name for this is
"specialist sub-sector".)

### Browse pages

- **Root browse page**: [gov.uk/browse](https://www.gov.uk/browse)
- **Top level browse page**: [gov.uk/browse/benefits](https://www.gov.uk/browse/benefits)

> - **Second level browse page**: [gov.uk/browse/benefits/entitlement](https://www.gov.uk/browse/benefits/entitlement)

Reviewer: Hmmm I did have some misgivings about removing this, as it could be quite useful to have an awareness of the different types of browse pages. But agreed it can be discovered at point of need. And the [vocabulary is already in the code](https://github.com/alphagov/collections/blob/c63209674bae27007a08ea172e7befb56fe500b5/config/routes.rb#L22-L24), so we're not particularly losing anything here.

- [search api](https://github.com/alphagov/search-api), provides:
    - latest changes for Topics
    - content tagged to a particular Topic, Mainstream browse page or Organisation

> - [email-alert-api](https://github.com/alphagov/email-alert-api), provides:

Reviewer: Good spot 👍

```

### Pact tests

Collections Organisations API has a pact with [GDS API Adapters](https://github.com/alphagov/gds-api-adapters).

> See [the guidance](https://docs.publishing.service.gov.uk/manual/pact-broker.html) for how to run and modify the tests.

Reviewer: Here's a new a11y suggestion from me: we now have two links with text as "the guidance", which is better than just "here" or "read more", but is still [quite vague when consumed in isolation](https://fae.disability.illinois.edu/rulesets/LINK_2/). This might be better:

```suggestion
Read [how to run and modify Pact tests](https://docs.publishing.service.gov.uk/manual/pact-broker.html).
```

Author: That's fair, particularly for the tech docs intro, which I'm expecting to be copied in future PRs for other frontend apps. I've amended both in the relevant commits.

How does it look now?

This is a public facing Ruby on Rails application that retrieves browse content from APIs and presents it. There is no underlying persistence layer, and (with the exception of the [campaign pages](#campaign-pages)) all content is retrieved from external sources.

### Content for taxon pages

> Content for taxon pages is returned by Search API based on content_ids for world taxonomy pages and content_ids and [supergroups](https://docs.publishing.service.gov.uk/document-types/content_purpose_supergroup.html) for all other taxonomy pages.

Reviewer: I don't think it would hurt to copy this info into the newly created taxonomy-pages.md.

Author: Sure, I agree it's got slightly different information in it than the separate doc. I've added a new commit to incorporate it.
